plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2855,7 +2858,7 @@ void producer_plugin_impl::produce_block() {
               "pending_block_state does not exist but it should, another plugin may have corrupted it");
 
    auto auth = chain.pending_block_signing_authority();
-   std::vector<std::reference_wrapper<const signature_provider_type>> relevant_providers;
+   std::vector<fc::crypto::signature_provider_ptr> relevant_providers;
```

Comment:
I don't think we should allow BLS keys for block signing. For that matter I don't think we should allow `ed` or `em` for block signing either. Those changes were before my time. We need to clamp down what is allowed to sign a block and also validate only those types are allowed during block validation.